### PR TITLE
feature/slider

### PR DESCRIPTION
**Pull request by SaldanRoman**
https://b24-lmffg5.bitrix24.ua/workgroups/group/8/tasks/task/view/32/

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -5,9 +5,21 @@
     <meta name="viewport"
           content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link href="styles.css" rel="stylesheet">
     <title>WineTime</title>
 </head>
 <body>
-
+<header>
+	 <div class="main-slider">
```
Comment (maradroman) [inline index.html]: class name is not in BEM methodology. expected "slider-main"

File: css/slider.css [hunk 2]
```diff
@@ -0,0 +1,159 @@
+.main-slider{
+    display: flex;
+    margin: 0 auto;
+    width: 100%;
+    overflow: hidden;
+    height: 500px;
+    position: relative;
+
```
Comment (maradroman) [inline css/slider.css:9]: unnecessary space

File: css/slider.css [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    transition: 300ms;
+}
+
+.slider-item-container{
+    margin: 0 auto;
+    width: 100%;
+    overflow: hidden;
+    height: 500px;
+    position: relative;
+
```
Comment (maradroman) [inline css/slider.css:46]: unnecessary space

File: css/slider.css [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    height: 100%;
+    display: flex;
+    position: absolute;
+    transition: 600ms;
+}
+
+.slider-item-background.left{
+    left: -100%;
+    transition: 600ms;
+
```
Comment (maradroman) [inline css/slider.css:68]: unnecessary space

File: css/slider.css [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+.slider-item-background.right{
+    left: 100%;
+    transition: 600ms;
+
```
Comment (maradroman) [inline css/slider.css:77]: unnecessary space

File: css/slider.css [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+.slider-item{
+    position: relative;
+    color: white;
+    text-shadow: 0px 0px 1px black;
+    display: flex;
+    width: 100%;
+    height: 100%;
+
```
Comment (maradroman) [inline css/slider.css]: unnecessary space

File: css/slider.css [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    display: flex;
+    width: 100%;
+    height: 100%;
+ 
+}
+
+.slider-item.right{
+    justify-content: flex-end;
+    margin: 4% 10% 0% 0%;
+
```
Comment (maradroman) [inline css/slider.css:96]: unnecessary space

File: css/slider.css [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+.slider-item-content-button.red{
+    background-color: #e5343e;
+}
+
+.slider-item-content-button.blue{
+    background-color: #9941ef;
+}
+
+.slider-item-content-button:hover {
+    background-color: rgb(252, 165, 60);
```
Comment (maradroman) [inline css/slider.css]: hex code expected, #fca53c

**Review by maradroman — COMMENTED** [review]
commented mistakes, please fix

File: css/slider.css [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+.slider-item-buttons{
+    display: flex;
+    align-items: center;
+    height: 100%;
+    position: absolute;
+    z-index: 1;
+    opacity: 0;
+    transition: 300ms;
```
Comment (romanivna) [inline css/slider.css]: .3s (leading zero is deprecated)

File: css/slider.css [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+.slider-item-buttons button{
+    border: none;
+    background-color: lightgrey;
+    transition: 300ms;
```
Comment (romanivna) [inline css/slider.css]: .3s

File: css/slider.css [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+    border-radius: 3px;
+    font-size: 1.7rem;
+    opacity: 0.5;
```
Comment (romanivna) [inline css/slider.css]: .5

**Review by OlecsandrStelmakh — DISMISSED** [review]

File: css/slider.css [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+    height: 100%;
+    position: absolute;
+    z-index: 1;
+    opacity: 0;
+    transition: 300ms;
+}
+
+.slider-item-buttons button{
+    border: none;
+    background-color: lightgrey;
```
Comment (maradroman) [inline css/slider.css]: hex code expected, #d3d3d3

File: css/slider.css [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+
+.slider-item-background{
+    background-size: cover;
+    background-position: center;
+    box-sizing: border-box;
+    width: 100%;
+    height: 100%;
+    display: flex;
+    position: absolute;
+    transition: 600ms;
```
Comment (maradroman) [inline css/slider.css]: .6s

File: js/slider.js [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+        } ,timeCount);
+        timeCount+=50;
+    }
+
+}
+
+let sliderInterval = setInterval(sliderNext, 6000)
+
+function hideText(){
+    let timeCount = 50;
```
Comment (maradroman) [inline js/slider.js]: redundant variable (unused)

**Review by OlecsandrStelmakh — APPROVED** [review]

**Review by OlecsandrStelmakh — DISMISSED** [review]

File: js/slider.js [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+    for (let i = itemTextArr.length - 1; i >= 0; i--) {
+        setTimeout(function() {
+            itemTextArr[i].classList.add('center')
+        }, timeCount);
+        timeCount += 50;
+    }
+}
+
+function hideText() {
+    let timeCount = 50;
```
Comment (romanivna) [inline js/slider.js]: remove unnecessary variable

Comment (SaldanRoman) [reply js/slider.js]: done !)

File: index.html [hunk 16]
```diff
@@ -5,9 +5,21 @@
     <meta name="viewport"
           content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link href="styles.css" rel="stylesheet">
     <title>WineTime</title>
 </head>
 <body>
-
+<header>
+	 <div class="slider-main">
```
Comment (romanivna) [inline index.html]: slider

File: index.html [hunk 17]
```diff
@@ -5,9 +5,21 @@
...
+            <div class="slider-item-buttons left">
```
Comment (romanivna) [inline index.html]: slider-button-wrapper slider-button-wrapper--left

File: index.html [hunk 18]
```diff
@@ -5,9 +5,21 @@
...
+                <button onclick="sliderPrevious()" id="mainSliderButtonLeft">&#10094</button>
```
Comment (romanivna) [inline index.html]: slider-button

File: index.html [hunk 19]
```diff
@@ -5,9 +5,21 @@
...
+            </div>  
+            <div class="slider-item-container"></div>
```
Comment (romanivna) [inline index.html]: slider-container

Comment (romanivna) [reply index.html]: > slider-container-item
      > slider-container-item-text
             >slider-container-item-text-title

File: css/slider.css [hunk 20]
```diff
@@ -0,0 +1,204 @@
...
+    display: flex;
+    align-items: center;
+    height: 100%;
+    position: absolute;
+    z-index: 1;
+    opacity: 0;
+    transition: .3s;
+}
+
+.slider-item-buttons button{
```
Comment (romanivna) [inline css/slider.css]: .slider-button

File: js/slider.js [hunk 21]
```diff
@@ -0,0 +1,152 @@
+const arrayOfSliderProd = [{
```
Comment (romanivna) [inline js/slider.js]: move to json

Comment (romanivna) [reply js/slider.js]: translate to english

File: js/slider.js [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+    imgSrc: 'img/slider/4.jpg',
+    title: 'Вино Provetto',
+    subtitle: 'Проветто Спуманте Бьянко Брют',
+    description: 'Глубинный и насыщенный вкус ',
+    buttonUrl: 'https://winebutik.com.ua/drinks/category/wines/brand/sassicaia',
+    buttonColor: 'blue',
+    position: 'right'
+}]
+
+function generatediv(ppclass, index, obj) {
```
Comment (romanivna) [inline js/slider.js]: generateSlide / Block / sliderBlock / sliderElement

Comment (romanivna) [reply js/slider.js]: generateAndAddToHtmlSlideBlock

Comment (romanivna) [reply js/slider.js]: ppclass???? positionClass

Comment (romanivna) [reply js/slider.js]: index > currentSlideIndex

Comment (romanivna) [reply js/slider.js]: obj => slideDataObj

File: js/slider.js [hunk 23]
```diff
@@ -0,0 +1,152 @@
...
+    const backgroundDiv = document.createElement('div');
+    backgroundDiv.setAttribute('class', 'slider-item-background');
+    backgroundDiv.setAttribute('style', `background-image:url(${obj.imgSrc})`);
+    backgroundDiv.setAttribute('data-index', index);
+    backgroundDiv.classList.add(ppclass);
+    //position of slide (left current right)
+    const sliderItem = document.createElement('div');
```
Comment (romanivna) [inline js/slider.js]: wrapper

File: js/slider.js [hunk 24]
```diff
@@ -0,0 +1,152 @@
...
+    sliderItem.setAttribute('class', 'slider-item');
+    sliderItem.classList.add(obj.position)
+    // contents position in slide ( center right)
+    const sliderItemContent = document.createElement('div');
```
Comment (romanivna) [inline js/slider.js]: textWrapper

File: js/slider.js [hunk 25]
```diff
@@ -0,0 +1,152 @@
...
+    sliderItemContent.setAttribute('class', 'slider-item-content');
+    const sliderItemContentTextTitle = document.createElement('p');
```
Comment (romanivna) [inline js/slider.js]: title

File: js/slider.js [hunk 26]
```diff
@@ -0,0 +1,152 @@
...
+    const sliderItemContentTextDescr = document.createElement('p');
+    sliderItemContentTextDescr.setAttribute('class', 'slider-item-content-text-descr');
+    sliderItemContentTextDescr.innerText = `${obj.description}`;
+    //description
+    const sliderItemContentButton = document.createElement('a');
+    sliderItemContentButton.setAttribute('class', 'slider-item-content-button')
+    sliderItemContentButton.setAttribute('href', obj.buttonUrl)
+    sliderItemContentButton.classList.add(obj.buttonColor)
+    //buttonColor
+    sliderItemContentButton.innerText = 'ПОПРОБОВАТЬ';
```
Comment (romanivna) [inline js/slider.js]: translate

File: js/slider.js [hunk 27]
```diff
@@ -0,0 +1,152 @@
...
+    sliderItemContent.appendChild(sliderItemContentButton);
+
+    sliderItem.appendChild(sliderItemContent)
+
+    backgroundDiv.appendChild(sliderItem)
+    return backgroundDiv
+}
+;
+(function() {
+    window.sliderContainer = document.querySelector('.slider-item-container');
```
Comment (romanivna) [inline js/slider.js]: fetch().then( () => geberatekhgkhkg)

File: js/slider.js [hunk 28]
```diff
@@ -0,0 +1,152 @@
...
+    sliderContainer.appendChild(generatediv('left', 3, arrayOfSliderProd[3]));
```
Comment (romanivna) [inline js/slider.js]: last index of array

Comment (romanivna) [reply js/slider.js]: const lastIndex = arrayOfSliderProd.length - 1;

File: js/slider.js [hunk 29]
```diff
@@ -0,0 +1,152 @@
...
+    sliderContainer.appendChild(generatediv('current', 0, arrayOfSliderProd[0]));
+    sliderContainer.appendChild(generatediv('right', 1, arrayOfSliderProd[1]));
+}
+)()
+
+function showText() {
+    let timeCount = 50;
```
Comment (romanivna) [inline js/slider.js]: rename

File: js/slider.js [hunk 30]
```diff
@@ -0,0 +1,152 @@
...
+    const itemTextArr = document.querySelectorAll('.slider-item-content p, .slider-item-content a');
```
Comment (romanivna) [inline js/slider.js]: .slider-item-content-text-subtitle
not a => set it a class

File: js/slider.js [hunk 31]
```diff
@@ -0,0 +1,152 @@
...
+    sliderItemContent.appendChild(sliderItemContentTextDescr);
+    sliderItemContent.appendChild(sliderItemContentButton);
+
+    sliderItem.appendChild(sliderItemContent)
+
+    backgroundDiv.appendChild(sliderItem)
+    return backgroundDiv
+}
+;
+(function() {
```
Comment (romanivna) [inline js/slider.js]: move to end

File: js/slider.js [hunk 32]
```diff
@@ -0,0 +1,152 @@
...
+    const leftImgIndex = getNextIndex(leftImg, 'left');
+    sliderContainer.appendChild(generatediv('left', leftImgIndex, arrayOfSliderProd[leftImgIndex]));
+    hideText();
+    setTimeout(showText, 300);
+
+}
+sliderPrevious();
+let mainsliderInterval = setInterval(sliderNext, 6000);
+
+document.querySelector('#mainSliderButtonLeft').addEventListener('click', function() {
```
Comment (romanivna) [inline js/slider.js]: getElementById

File: js/slider.js [hunk 33]
```diff
@@ -0,0 +1,152 @@
...
+    clearInterval(mainsliderInterval);
```
Comment (romanivna) [inline js/slider.js]: 146 and 147 strings -> move to function

Comment (romanivna) [reply js/slider.js]: resetInterval()

File: js/slider.js [hunk 34]
```diff
@@ -0,0 +1,152 @@
...
+    mainsliderInterval = setInterval(sliderNext, 6000);
+});
+document.querySelector('#mainSliderButtonRight').addEventListener('click', function() {
+    clearInterval(mainsliderInterval);
```
Comment (romanivna) [inline js/slider.js]: clickToButton(position) {
if (position === 'left') { sliderPrevious() }
else sliderNext()
resetInterval()
}

File: index.html [hunk 35]
```diff
@@ -5,9 +5,21 @@
...
+            <div class="slider-item-buttons right">
+                <button onclick="sliderNext()" id="mainSliderButtonRight">&#10095</button>
```
Comment (romanivna) [inline index.html]: onclick=clickToButton('left' || 'right')

File: css/slider.css [hunk 36]
```diff
@@ -0,0 +1,204 @@
+.slider-main{
```
Comment (romanivna) [inline css/slider.css]: add Prettier plugin for your IDE and create hotkey for it. use it every time when you save the file.
between selector and brace should be space.

File: css/slider.css [hunk 37]
```diff
@@ -0,0 +1,204 @@
...
+.slider-item-buttons.right{
+    right: 0;
+}
+
+.slider-main:hover .slider-item-buttons{
+    opacity: 1;
+    transition: .3s;
+}
+
+.slider-item-buttons button:hover{
```
Comment (romanivna) [inline css/slider.css]: set button his own class and use it instead of this construnction

File: css/slider.css [hunk 38]
```diff
@@ -0,0 +1,204 @@
...
+    color: #ffffff;
```
Comment (romanivna) [inline css/slider.css]: use short hex names (fff instead of ffffff)

File: css/slider.css [hunk 39]
```diff
@@ -0,0 +1,204 @@
...
+
+.slider-item-background.current{
+    left: 0;
+    transition: .6s;
+}
+
+.slider-item{
+    position: relative;
+    color: #ffffff;
+    text-shadow: 0px 0px 1px black;
```
Comment (romanivna) [inline css/slider.css]: when value is 0 - there couldn't be any measurements. 0% and 0px it's a zero.
so don't write measure.
use just "0"

File: css/slider.css [hunk 40]
```diff
@@ -0,0 +1,204 @@
...
+    display: flex;
+    width: 100%;
+    height: 100%;
+ }
+
+.slider-item.right{
+    justify-content: flex-end;
+    margin: 4% 10% 0% 0%;
```
Comment (romanivna) [inline css/slider.css]: same, just 0
look throw all css and remove measures after 0

File: js/slider.js [hunk 41]
```diff
@@ -0,0 +1,85 @@
...
+    const currentIndex = +el.getAttribute('data-index');
+    if (direction === 'right') {
+        return arrayOfSliderProd[currentIndex + 1] ? currentIndex + 1 : 0
+    }
+    return arrayOfSliderProd[currentIndex - 1] ? currentIndex - 1 : arrayOfSliderProd.length - 1
+}
+
+let changSlideinterval = setInterval(changSlide, 6000);
+
+function changSlide(direction){
```
Comment (OlecsandrStelmakh) [inline js/slider.js]: space before brackets

File: js/slider.js [hunk 42]
```diff
@@ -0,0 +1,85 @@
...
+    const leftImg = document.querySelector('.slider-content--left');
+    const curentImg = document.querySelector('.slider-content--current');
+    const rightImg = document.querySelector('.slider-content--right');
+    if(direction === 'left'){
+        leftImg.remove();
+        curentImg.classList.replace('slider-content--current', 'slider-content--left');
+        rightImg.classList.replace('slider-content--right', 'slider-content--current');
+        slider.appendChild(generateSlide('slider-content--right', getNextIndex(rightImg, 'right'), 
+            arrayOfSliderProd[getNextIndex(rightImg, 'right')]));
+    }else{
```
Comment (OlecsandrStelmakh) [inline js/slider.js]: space before brackets

**Review by OlecsandrStelmakh — DISMISSED** [review]

File: css/slider.css [hunk 43]
```diff
@@ -0,0 +1,187 @@
...
+
+.slider-content-text--right{
+    justify-content: flex-end;
+}
+
+.slider-content-text--right .slider-content-text{
+    margin: 30px 130px 0 0 ;
+}
+
+.slider-content--left{
```
Comment (romanivna) [inline css/slider.css]: SPACES!!!!!

File: js/slider.js [hunk 44]
```diff
@@ -0,0 +1,85 @@
+let arrayOfSliderProd;
+fetch('./jsons/slider.json').then(respons => respons.json()).then(data => {arrayOfSliderProd = data; createSlider(); changSlide()});
```
Comment (romanivna) [inline js/slider.js]: response

Comment (romanivna) [reply js/slider.js]: and please makeeach step on separate line.add prettier.

Comment (romanivna) [reply js/slider.js]: when you call function - this function should be close as soon as possible to place where it called.

File: js/slider.js [hunk 45]
```diff
@@ -0,0 +1,85 @@
...
+    sliderText.appendChild(subTitleText);
+    sliderText.appendChild(descriptionText)
+    sliderText.appendChild(linkText);
+    slideContent.appendChild(sliderText);
+    return slideContent
+}
+
+function createSlider(){
+    window.slider = document.querySelector('.slider-content-wrapper');
+    slider.appendChild(generateSlide('slider-content--left', arrayOfSliderProd.length-1, arrayOfSliderProd[arrayOfSliderProd.length-1]));
```
Comment (romanivna) [inline js/slider.js]: arrayOfSliderProd.length-1 move to variable

File: js/slider.js [hunk 46]
```diff
@@ -0,0 +1,85 @@
...
+    slider.appendChild(generateSlide('slider-content--current', 0, arrayOfSliderProd[0]))
+    slider.appendChild(generateSlide('slider-content--right', 1, arrayOfSliderProd[1]))
+}
+
+function textMoveIn(){
+    let timeoutValue= 50;
```
Comment (romanivna) [inline js/slider.js]: rename

File: js/slider.js (hunk 44 above)
Comment (romanivna) [reply js/slider.js]: `fetch('./jsons/slider.json')
  .then(respons => respons.json())
  .then(data => {
     arrayOfSliderProd = data;
     createSlider(); 
     changSlide()
});`

**Review by maradroman — DISMISSED** [review]

**Review by OlecsandrStelmakh — DISMISSED** [review]

**Review by OlecsandrStelmakh — APPROVED** [review]

**Review by romanivna — DISMISSED** [review]

**Review by OlecsandrStelmakh — CHANGES_REQUESTED** [review]
on the phone version the problem with the pictures (invisible bottles), and in general it was not bad that the text appeared more slowly

**Review by OlecsandrStelmakh — DISMISSED** [review]

**Review by OlecsandrStelmakh — DISMISSED** [review]

File: css/slider.css [hunk 47]
```diff
@@ -0,0 +1,207 @@
...
+
+.slider-buttons:hover {
+  opacity: 1;
+  transition: 0.3s;
+  color: #fff;
+  background-color: #ff8c00;
+}
+
+.slider-buttons-wrapper--left {
+  left: 0.5%;
```
Comment (romanivna) [inline css/slider.css]: .5s
make settings or prettier with this rule or add sass-lint also

**Review by romanivna — DISMISSED** [review]

File: styles.css [hunk 48]
```diff
@@ -1,31 +1,34 @@
 @import "css/benefits.css";
+
```
Comment (romanivna) [inline styles.css]: we don't need empty lines between imports.
empty line should be after all imports before file content
remove please

**Review by romanivna — APPROVED** [review]

**Pull request merged**